internal/cli/usercmd/service.go

```diff
@@ -447,6 +447,17 @@ func (c *EpinioClient) ServicePortForward(ctx context.Context, serviceName strin
 		return err
 	}
 
+	/*
+		We use ServiceList API to check if we can establish a connection with the Epinio backend service.
+		In this scenario we are not really interested into retrieve details about the actual error returned;
+		Most common scenarios here are that the backend service is not up and running or that the client is not
+		logged it.
+	*/
+	_, err := c.API.ServiceList(c.Settings.Namespace)
+	if err != nil {
+		return errors.New("Cannot establish a connection with Epinio. Check both if Epinio is up and if the client is logged in.")
+	}
```

Comment:
From a recent refactor for logic errors we are now getting an `APIError` that can be used to check what kind of error was returned.
It's also better to `Wrap` the error instead of creating a new one, to get all the stack trace.


```suggestion
	/*
		We use ServiceList API to check if we can establish a connection with the Epinio backend service.
		In this scenario we are not really interested into retrieve details about the actual error returned.
		Most common scenarios here are that the backend service is not up and running or that the client is not
		logged in.
	*/
	_, err := c.API.ServiceList(c.Settings.Namespace)
	if err != nil {
		epinioAPIError := &client.APIError{}
		if errors.As(err, &epinioAPIError) {
			statusCode := epinioAPIError.StatusCode
			if statusCode == http.StatusUnauthorized || statusCode == http.StatusForbidden {
				return errors.Wrap(err, "user is not authorized")
			}
		}
		return errors.Wrap(err, "getting services")
	}
```

Reply:
Actually, if we wrap the error (as I initially did) we would get this kind of message:

```
🚢  Executing port forwarding
Namespace: workspace
Service: test

❌  error port forwarding to service: Connection with backend failed: making the request: Get "/api/v1/namespaces/workspace/services": unsupported protocol scheme ""

```

I don't like this error because it is cryptic.
What are the most common scenarios of failure here?

1) Server not running
2) Client not authenticated

IMO, the user would prefer a clear human readable error message rather than a too much technical one when the 99.9% of failure scenarios are the 2 above.

No problem changing this btw! :grin:

Reply:
Not really cryptic but it could be nicer. At the moment on the CLI side we are just bubbling up the errors with all the stacktrace, and as you said it's not very nice from an end user perspective.
I would prefer to keep the full ugly stack trace at the moment, and maybe open an enhancement task to improve this.

With the improvement of the HTTP client it's now possible to get the proper status code, and the error, in an easy way, and this could be a way to show proper errors.

Actually the most common scenarios here are "real" failures. If you are trying to interact with an `epinio service port-forward` command I'm expecting the client to be already logged in, with an epinio server running. It's not probably the first command that you are going to execute. :smile:

Reply:
> I would prefer to keep the full ugly stack trace at the moment, and maybe open an enhancement task to improve this.

ok, I will do.
I will try also to implement a test.
Unfortunately, I really dunno if I will be able to address this for this week (so this could actually shift to 1.10).